### PR TITLE
Add python, OpenCV and python2 support to libfreenect

### PR DESCRIPTION
**Pull request by xionglingfeng**
Add options to compile libfreenect with OpenCV and Python

### CODE REVIEW

File: Formula/libfreenect.rb [hunk 1]
```diff
@@ -13,13 +13,33 @@ class Libfreenect < Formula
...
+  option :universal
+  option "with-opencv", "Build with OpenCV support"
+  option "with-opencv3", "Build with OpenCV 3 support"
+  option "with-python", "Build with Python support"
+  option "with-python3", "Build with Python 3 support"
+
   depends_on "cmake" => :build
   depends_on "libusb"
+  depends_on "opencv" if build.with? "opencv"
+  depends_on "opencv3" if build.with? "opencv3"
```
Comment (tdsmith) [inline Formula/libfreenect.rb]: These should look like `depends_on "homebrew/science/opencv" => :optional`

File: Formula/libfreenect.rb [hunk 2]
```diff
@@ -13,13 +13,33 @@ class Libfreenect < Formula
...
+  depends_on "python" if build.with? "python"
+  depends_on "python3" if build.with? "python3"
```
Comment (tdsmith) [inline Formula/libfreenect.rb]: These should look like `depends_on :python3 => :optional`.

If this just builds extension modules, just always build the python 2 modules and don't declare a :python dependency.

File: Formula/libfreenect.rb [hunk 3]
```diff
@@ -13,13 +13,33 @@ class Libfreenect < Formula
     sha256 "f714532e1b21365063746846544a340dac70cf0c5cc877a207dd17284ee100b7" => :mavericks
   end
 
+  option :universal
+  option "with-opencv", "Build with OpenCV support"
+  option "with-opencv3", "Build with OpenCV 3 support"
+  option "with-python", "Build with Python support"
+  option "with-python3", "Build with Python 3 support"
```
Comment (tdsmith) [inline Formula/libfreenect.rb]: Please remove all of the `option` calls (including `:universal`, unless there's a really compelling use case?); the `depends_on... => :optional` syntax does this for you.

**Review by tdsmith — CHANGES_REQUESTED** [review]

**Comment by tdsmith** [comment]
Thanks for the PR! Please let me know if you have any questions about the changes I requested.

**Comment by fxcoudert** [comment]
Closing this PR for now, as it has not been updated in response to review comments. Please reopen when addressing the comments.

**Pull request closed without merging**